src/commonMain/kotlin/me/alllex/parsus/token/RegexToken.kt

```diff
@@ -0,0 +1,71 @@
+package me.alllex.parsus.token
+
+import me.alllex.parsus.parser.GrammarContext
+
+/**
+ * A token that [matches] the input using a [regex].
+ */
+class RegexToken(
+    regex: Regex,
+    name: String? = null,
+    ignored: Boolean = false,
+) : Token(name, ignored) {
+
+    private val pattern: String = regex.pattern
+    private val regex: Regex = prependPatternWithInputStart(pattern, regex.options)
+//    private val matcher: Matcher = this.regex.toPattern().matcher("")
+    //private val matcher: RegexMatcher = this.regex.toMatcher()
+
+    private fun prependPatternWithInputStart(patternString: String, options: Set<RegexOption>): Regex {
+        return if (patternString.startsWith(inputStartPrefix))
+            patternString.toRegex(options)
+        else {
+//            val newlineAfterComments = if (RegexOption.COMMENTS in options) "\n" else ""
+//            val patternToEmbed = if (RegexOption.LITERAL in options) Regex.escape(patternString) else patternString
+//            ("$inputStartPrefix(?:$patternToEmbed$newlineAfterComments)").toRegex(options - RegexOption.LITERAL)
+          "$inputStartPrefix(?:$patternString)".toRegex(options)
+          // TODO is the warning correct?
+          //      > Unnecessary non-capturing group '(?:patternString)'
+        }
+    }
+
+    override fun match(input: CharSequence, fromIndex: Int): Int {
+        val result = regex.find(input, fromIndex) ?: return 0
+        return result.range.last + 1
+
+        //matcher.reset(input).region(fromIndex, input.length)
+        //if (!matcher.find()) return 0
+        //val end = matcher.end()
+        //return  end - fromIndex
+    }
```

Comment:
`Matcher` is JVM only, but it looks like its not necessary, and just using `Regex.find()` works?

Reply:
Hmm never mind - the tests in `:demo`: fail.

Reply:
All the tests pass now.

`find()` doesn't work with a `^` anchor, so instead I used trimmed the string: `input.drop(fromIndex)`. This is possibly less performant?